kuma/users/migrations/0018_user_subscriber_number.py

```diff
@@ -0,0 +1,30 @@
+# Generated by Django 2.2.11 on 2020-03-30 07:17
+
+from django.db import migrations, models
+
+
+def set_initial_subscriber_number(apps, schema_editor):
+    User = apps.get_model("users", "User")
+    number = 1
+    for user in User.objects.exclude(
+        stripe_customer_id="", subscriber_number__isnull=True
+    ):
+        user.subscriber_number = number
+        user.save()
+        number += 1
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("users", "0017_usersubscription"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="user",
+            name="subscriber_number",
+            field=models.PositiveIntegerField(blank=True, null=True),
```

Comment:
Alternative idea, make this a blank auto increment field, and then set it to its default value, which would simplify the updating code (aka outsource it to MySQLs data structures).

Tbh it's probably 100x eaiser to have this idea than to ORM-wrangle Django into making it work, so feel free to ignore it.

Reply:
I don't know what a "blank auto increment field" is. 
I'm hesitant to step outside the ORM if it's possible. My sticking to the ORM API you become more independent of the database vendor. 
Auto-incrementing things and stuff like sequences in PostgreSQL are generally scary because of transactions. If you manually maintain it instead you can make sure it doesn't skip.

Reply:
Wrt db vendor independentness: I haven't seen a single DB technology switch in later stage products yet, so I consider that ORM feature to be over-sold. But I also share your hope that MDN will someday run on PostgreSQL, so I'll take that point.

Wrt transaction failure: I don't foresee how 
```sql
UPDATE auth_users SET subscriber_no = NEXTVAL('subscriber_numbers') WHERE id = ?
```
could fail in a way that would bump the sequence. I.e. we wouldn't even need a transaction for use-case.

Though it probably works differently (for the worse 🙊) in MySQL which might make this more pain than gain indeed.

Reply:
I thought `NEXTVAL` was a Postgres-only thing since Postgres uses sequences as an underlying thing to generate primary key IDs.
My first iteration was to use something like this:
```python
with connection.cursor() as cursor:
    cursor.execute("""
        UPDATE auth_user 
        SET subscriber_number = (SELECT MAX(subscriber_number) FROM auth_user) 
        WHERE id=%s
    """, [user.id])
```
But you're not allowed to do edits and selects on the same table in MySQL.

Reply:
Hmm... 
```
django.db.utils.OperationalError: (1305, 'FUNCTION test_developer_mozilla_org.NEXTVAL does not exist')
```

Reply:
Oh wow, I wasn't aware of that limitation in MySQL.

And I also just learned that "Incorrect table definition; there can be only one auto column". So there's literally no other way to do it, than the way you did do it.

I just learned something. It's mostly that I miss postgres.

Reply:
And yup I was thining of PG when I said `NEXTVAL`, thought we were in a similar mindspace since you mentioned PG sequences before.

Reply:
I'm not entirely sure it's worth going into the trouble, considering how rarely people will become subscribers. So to have two people sign up within the same millisecond is sorry unlikely. 
But, before merging this, I need to think about the use of locking. There's no point locking the user's row because that's not where the risk lies. It's about the relationship to _other_ users. 
So, technically you should lock the whole _table_ but that's silly because why need with it for the other 99.9% when they don't care about this number. 
No, the best solution has to be to use a global lock specifically for this.